RH Docs 11367

Raghav2018 opened the pull request:
Created Java-based sample app for find() with scalar fields.

The sample app has four methods: 
1. Illustrates a simple equality match.
2. Illustrates an equality match on two or more fields.
3. Illustrates a range query on order total and makes use of projection.
4. Illustrates a range query by date and makes use of projection.

It also includes the unit tests for each of the above methods and one for connection to the cluster.

Raghav2018 committed:
Find queries with scalar fields

First Commit

Raghav2018 committed:
Added unit test case for connection to cluster

Second commit

Raghav2018 committed:
Changed method names to more readable format

Changes based on common review comments

Raghav2018 committed:
Worked on review comments

- Handled edge cases
- Added more robust unit tests

Raghav2018 commented:
@shannonbradshaw I have worked on the review comments. Please review my new code.

shannonbradshaw commented on FindWithScalarFields/src/FindWithScalarFields.java:
```diff
@@ -0,0 +1,207 @@
+		}
+		else
+		{
+			MongoClientURI clientUri = new MongoClientURI(connectionString);
```
Not sure it makes sense to create a variable for the uri. What made you decide to do that rather than simply embedding the call to MongoClientURI() in the call to the constructor for MongoClient on the next line. This is a coding style comment, fyi.

Raghav2018 replied:
I just wanted to stay inline with the already available documentation on mongoDB website. https://docs.atlas.mongodb.com/driver-connection/#java-driver-example

Shall I change this by excluding MongoClientUri creation?

shannonbradshaw commented on FindWithScalarFields/src/FindWithScalarFields.java:
```diff
@@ -0,0 +1,207 @@
+	 * @param total: Order total to filter documents.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	public List<Document> GetOrdersByPostalCodeAndTotal(String connectionString, int postalcode, Double total){
```
Why did you choose this particular query? Do you believe it is common that people who use our system would frequently want to search both those those fields.

shannonbradshaw replied:
Also searching by total except in a range query would be very uncommon.

shannonbradshaw replied:
Need different examples here. Need examples that are easily to complete, but provide readers with a plausible scenario.

shannonbradshaw commented on FindWithScalarFields/src/FindWithScalarFields.java:
```diff
@@ -0,0 +1,207 @@
+	
+	/**
+	 * Retrieves documents where orderPlaced is between startDate and endDate.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
```
Great example!

shannonbradshaw commented:
Done with another review pass.

Raghav2018 committed:
Worked on further review comments

- Multiple fields equality match: Changed it to a plausible situation
where user queries 'Orders' placed on a particular date to a particular
postal code.

Raghav2018 commented:
@shinyou21 Worked on Shannon's review comments. Please give it another review pass.

shinyou21 commented on FindWithScalarFields/src/FindWithScalarFields.java:
```diff
@@ -0,0 +1,207 @@
+			}
+			catch (Exception e) {
+				System.out.println("Exception occured");
+				System.out.println("Deatils:");
```
Just a typo on Details. repeated in all Exception below

Raghav2018 committed:
Rectified the typo

Raghav2018 commented:
@shinyou21 Fixed the typos. Let me know if I can merge this pull request

shinyou21 reviewed (approved):
Looks Good To Me
